ComponentKit/Core/Scope/CKComponentScopeTypes.h

```diff
@@ -16,3 +16,36 @@ typedef int32_t CKComponentScopeHandleIdentifier;
 typedef int32_t CKComponentScopeRootIdentifier;
 
 typedef std::unordered_multimap<CKComponentScopeHandleIdentifier, id (^)(id)> CKComponentStateUpdateMap;
+
+@protocol CKScopedComponent;
+@protocol CKScopedComponentController;
+
+/**
+ Enumerator blocks allow a consumer to enumerate over all of the components or controllers that matched a predicate.
+ */
+typedef void (^CKComponentScopeEnumerator)(id<CKScopedComponent>);
```

Comment:
I agree with you both about the renaming and having a separate file - I'll open a task and get to this later.